### ISSUE TITLE
Pillow in AWS Lambda not working properly

### ISSUE THREAD

**Issue by yasirtrimulabs** [python-pillow/Pillow#7648]
### What did you do?

I have a use case where I merge some images and write some text to create thumbnails.

The images are croping and merging fine but when I try to use ImageFont and load a font from S3 bucket it gives me the following error:

```
/lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by /opt/python/PIL/../Pillow.libs/libpng16-78d422d5.so.16.40.0)"
```

### What did you expect to happen?

Thumbnail with text

### What actually happened?

Throws this error:

```
/lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by /opt/python/PIL/../Pillow.libs/libpng16-78d422d5.so.16.40.0)"
```

### What are your OS, Python and Pillow versions?

* OS: AWS Lambda using Python 3.9
* Python: 3.9
* Pillow: 10.1.0 

```python
       # ... rest of the code
        # Use ImageDraw to add the text
        draw = ImageDraw.Draw(result_image)

        # Set the font and size
        font_size = 80
        font_path = "s3://[BUCKET_NAME]/Montserrat-Bold.ttf"
        font = ImageFont.truetype(font_path, font_size)  # <--- This causes the issue

        # Set the position for the text
        text_position = (320, 432)  # Adjust the coordinates as needed

        # Set the color for the text
        text_color = (255, 255, 255)  # White color

        # Draw the text on the result_image
        draw.text(text_position, text, font=font, fill=text_color)

       # ... rest of the code
```

### Additional Info
I am creating a layer for lambda using docker.
I have tried opencv-python as suggested in issues for other packages.
Also, there were some other suggestions on some forums but they didn't work or were not applicable in this scenario.
Just wanted to mention that I have searched alot for this but still couldn't find anything that worked.

**Comment by nulano** [comment]
How did you install Pillow? I guess it might be easiest if you just post your `Dockerfile`.

**Comment by yasirtrimulabs** [comment]
@nulano here's the `Dockerfile`

```
# Use the official Python runtime as the base image
FROM python:3.9

# Set the working directory in the container
WORKDIR /app

# Install any necessary dependencies
RUN apt-get update \
    && apt-get install -y zip \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*  /var/cache/apt/archives/*

# Copy the requirements file to the working directory
COPY requirements.txt .

# Upgrade pip
RUN pip install --upgrade pip

# Install the Python packages listed in requirements.txt
RUN pip install -r requirements.txt -t /opt/python/

# Set the CMD to clean up unnecessary files before zipping
CMD cd /opt && find /opt/python/ -type d -name "__pycache__" -exec rm -rf {} + \
    && find /opt/python/ -type f -name "*.pyc" -exec rm -f {} + \
    && find /opt/python/ -type d -name "tests" -exec rm -rf {} + \
    && zip -r9 /app/dependencies.zip .

```

Here's the `create-layer.sh` script which creates the zip
```
#!/bin/bash

# Set the directory where the Dockerfile and requirements.txt are located
DIRECTORY="$(pwd)"

# Change it as per your requirement
LAYER_NAME="dependencies"

# Build the Docker image
docker build -t lambda-layer "$DIRECTORY"

# Run the Docker container to create the layer
docker run --name lambda-layer-container -v "$DIRECTORY:/app" lambda-layer

# Create layers directory, if not created
mkdir -p layers

# Move the zip file in layers directory
mv "$DIRECTORY/$LAYER_NAME.zip" "$DIRECTORY/layers/$LAYER_NAME.zip"

# Stop the container
docker stop lambda-layer-container

# Remove the running container
docker rm lambda-layer-container

# Cleanup: remove the Docker image
docker rmi --force lambda-layer

```

Here's the requirements.txt file

```
wand
boto3
Pillow
requests
```

I have followed this guide initially https://medium.com/simform-engineering/creating-lambda-layers-made-easy-with-docker-a-developers-guide-3bcfcf32d7c3

**Comment by nulano** [comment]
So you are using a Debian-based container to install Python, then copying the files to AWS Lambda?
It does not surprise me that it doesn't work for you, since according to [this](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html), the AWS Lambda system is based on Amazon Linux 2.

Comparing what happens if you run pip in each container, you can see that it uses a different wheel on each system.

The `python:3.9` image (based on Debian Bookworm) uses the `manylinux_2_28_x86_64` wheel:
```
nulnao$ docker run --rm -it python:3.9 sh
# python3 -m pip install Pillow
Collecting Pillow
  Downloading Pillow-10.1.0-cp39-cp39-manylinux_2_28_x86_64.whl (3.6 MB)
[snip]
```

The [`pythonpillow/amazon-2-amd64:main` image](https://github.com/python-pillow/docker-images/blob/main/amazon-2-amd64/Dockerfile) (`amazonlinux:2` with Python 3.9) uses the (older) `manylinux_2_17_x86_64` wheel:
```
nulano$ docker run --rm -it pythonpillow/amazon-2-amd64:main sh
sh-4.2$ python3 -m pip install Pillow
Defaulting to user installation because normal site-packages is not writeable
Collecting Pillow
  Downloading Pillow-10.1.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.5 MB)
[snip]
```

So it might be better if you can build your package on Amazon Linux 2, perhaps using something similar to the [`pythonpillow/amazon-2-amd64:main` image](https://github.com/python-pillow/docker-images/blob/main/amazon-2-amd64/Dockerfile).

**Comment by yasirtrimulabs** [comment]
Hi @nulano I finally fixed this by using a previous version of Pillow `8.3.1`
It took a lot of time but I just tested multiple and found this one to be working

**Comment by nulano** [comment]
If you'd like to use a more recent version of Pillow (there have been many fixes, including security ones, since 8.3.1), and have to use Debian, you can also add `--platform manylinux_2_17_x86_64 --only-binary :all:` to the pip install command. The following should work:

```Dockerfile
RUN pip install --platform manylinux_2_17_x86_64 --only-binary :all: -r requirements.txt -t /opt/python/
```

**Comment by yasirtrimulabs** [comment]
> If you'd like to use a more recent version of Pillow (there have been many fixes since 8.3.1, including security ones), and have to use Debian to download it, you can also add `--platform manylinux_2_17_x86_64 --only-binary :all:` to the pip install command. The following should work:
> 
> ```dockerfile
> RUN pip install --platform manylinux_2_17_x86_64 --only-binary :all: -r requirements.txt -t /opt/python/
> ```

Thanks I will try and let you know
